graph-query-ir/src/main/java/oracle/pgql/lang/spatial/Point2D.java

```diff
@@ -40,6 +41,35 @@ public Point2D(double x, double y, double m) {
     this.m = m;
   }
 
+  /**
+   * Create a Point2D object from a well-known text representation
+   * @param wktPoint well-known text representation of a point value
+   * @return Point2D object
+   */
+  public static Point2D fromWkt(String wktPoint) {
+    String prefix = wktPoint.substring(0, wktPoint.indexOf('('));
+    String values = wktPoint.substring(wktPoint.indexOf('(') + 1, wktPoint.indexOf(')'));
```

Comment:
Here we may want to safeguard from any exceptions like IndexOutOfBoundsException.
Maybe you can create a new Exception class like `WktParseException` and throw `new WktParseException(wktPoint + " is not in the right format")` if `wktPoint.indexOf('(')` returns `-1`. There may be some more cases below.